pkg/apis/aws/validation/shoot_test.go

```diff
@@ -165,10 +168,65 @@ var _ = Describe("Shoot validation", func() {
 				))
 			})
 		})
+
+		Describe("#ValidateWorkersUpdate", func() {
+			It("should pass because workers are unchanged", func() {
+				newWorkers := copyWorkers(workers)
+				errorList := ValidateWorkersUpdate(workers, newWorkers, field.NewPath("workers"))
+
+				Expect(errorList).To(BeEmpty())
+			})
+
+			It("should allow adding workers", func() {
+				newWorkers := append(workers[:0:0], workers...)
+				workers = workers[:1]
+				errorList := ValidateWorkersUpdate(workers, newWorkers, field.NewPath("workers"))
+
+				Expect(errorList).To(BeEmpty())
+			})
+
+			It("should allow adding a zone to a worker", func() {
+				newWorkers := copyWorkers(workers)
+				newWorkers[0].Zones = append(newWorkers[0].Zones, "another-zone")
+				errorList := ValidateWorkersUpdate(workers, newWorkers, field.NewPath("workers"))
+
+				Expect(errorList).To(BeEmpty())
+			})
+
+			It("should forbid removing a zone from a worker", func() {
+				newWorkers := copyWorkers(workers)
+				newWorkers[1].Zones = newWorkers[1].Zones[1:]
+				errorList := ValidateWorkersUpdate(workers, newWorkers, field.NewPath("workers"))
+
+				Expect(errorList).To(ConsistOf(
+					PointTo(MatchFields(IgnoreExtras, Fields{
+						"Type":  Equal(field.ErrorTypeInvalid),
+						"Field": Equal("workers[1].zones"),
+					})),
+				))
+			})
+
+			It("should forbid changing the zone order", func() {
+				newWorkers := copyWorkers(workers)
+				newWorkers[0].Zones[0] = workers[0].Zones[1]
+				newWorkers[0].Zones[1] = workers[0].Zones[0]
+				errorList := ValidateWorkersUpdate(workers, newWorkers, field.NewPath("workers"))
+
+				Expect(errorList).To(ConsistOf(
+					PointTo(MatchFields(IgnoreExtras, Fields{
+						"Type":  Equal(field.ErrorTypeInvalid),
+						"Field": Equal("workers[0].zones"),
+					})),
+				))
+			})
+		})
 	})
 })
 
```

Comment:
Can you add one more test that checks that adding a new zone while changing the existing zones is not allowed?